rustual-boy-core/src/v810.rs

```diff
@@ -993,15 +993,24 @@ impl V810 {
             return;
         }
 
-        self.enter_exception(exception_code);
+        let interrupt_level = (exception_code as usize >> 4) & 0x0f;
+        if interrupt_level < self.psw_interrupt_mask_level {
+            return;
+        }
+
+        self.enter_exception(exception_code, interrupt_level);
     }
 
-    fn enter_exception(&mut self, exception_code: u16) {
+    fn enter_exception(&mut self, exception_code: u16, mut interrupt_level: usize) {
```

Comment:
nit: `mut` args should be avoided where possible. In this case, `self.psw_interrupt_mask_level = if interrupt_level < 15 { interrupt_level + 1 } else { 15 };` might have been better, but it's certainly not a dealbreaker :)